### PR TITLE
Add filter options

### PR DESCRIPTION
**Pull request by Sharkesm**
**Implementation:** 
Users will be able to access the exposed `filter option` methods to add or remove filtering options based on their choice.

The filter option is comprised of: 
- `countryName`
- `countryDialCode`
- `countryCode`

Filter manipulation methods is comprised of: 
- `CountryManager.shared.addFilter(_:)`
- `CountryManager.shared.remvoeFilter(_:)`
- `CountryManager.shared.clearAllFilters()`

**Note:** 
By default, `Country Picker` scene is configured with `CountryFilterOptions.countryName` filter option. Incase the user invokes `CountryManager.clearAllFilters()` method. The default filter option will be set to `CountryFilterOptions.countryName`.

### CODE REVIEW

File: CountryPicker/CountryPicker/CountryManager.swift [hunk 1]
```diff
@@ -43,8 +54,21 @@ open class CountryManager {
...
+    
+    internal var lastCountrySelected: Country?
+    
+    /// Default country filter option
+    internal let defaultFilter: CountryFilterOptions = .countryName
+    
+    /// Exposed country filter options and should be configured by user
+    ///
+    /// - Note: By default, countries can be filtered by there country names
+    internal var filters: [CountryFilterOptions] = [.countryName]
```
Comment (SURYAKANTSHARMA) [inline CountryPicker/CountryPicker/CountryManager.swift]: May be we can use `Set` as better representable data structure here. so only we need to `filter.instert` will take care of duplication

File: CountryPicker/CountryPicker/CountryManager.swift [hunk 2]
```diff
@@ -66,6 +90,48 @@ open class CountryManager {
...
+        
+        var filterSets = Set(self.filters)
+        
+        if filterSets.contains(filter) {
+            return
+        }
+        
+        filterSets.insert(filter)
+        self.filters.removeAll()
+        self.filters.append(contentsOf: Array(filterSets))
```
Comment (SURYAKANTSHARMA) [inline CountryPicker/CountryPicker/CountryManager.swift]: We can directly `filters.instert(filter)` after changing data type to `Set` [insert doc](https://developer.apple.com/documentation/swift/set/3128848-insert)

File: CountryPicker/CountryPicker/CountryManager.swift [hunk 3]
```diff
@@ -9,6 +9,19 @@
 import Foundation
 import UIKit
 
+
+/// Country filtering options
+public enum CountryFilterOptions {
```
Comment (SURYAKANTSHARMA) [inline CountryPicker/CountryPicker/CountryManager.swift]: `CountryFilterOptions` look like array may be `CountryFilterOption` will be better name for enum. Any thought ?

Comment (Sharkesm) [reply CountryPicker/CountryPicker/CountryManager.swift]: Yeah makes more sense with singular context.

File: CountryPicker/CountryPicker/CountryManager.swift (hunk 1 above)
Comment (Sharkesm) [reply CountryPicker/CountryPicker/CountryManager.swift]: Thanks for pointing that out. That will definitely solve that duplication issue.

**Pull request merged**